autolab/Autolab
# Version dropdown: jump to same file if it exists

## Description
- When switching versions via the version dropdown, jump to the same file if it exists
- Add an indicator to the dropdown showing which versions have the file (and add a tooltip on hover)
- Switch from `url_for` to `_path` methods where applicable

## Motivation and Context

Currently, when the version dropdown is used, the first file of the submission is always loaded, even if the same file exists in the submission. Furthermore, there is no indication which versions contain the same file.

This makes it more tedious to jump between versions of the same file, as an instructor must instead use the "next version" and "previous version" buttons to go through the versions one at a time.

Closes #1729

## How Has This Been Tested?
Create a lab and make multiple tar submissions, where each tar differs in some files (might be easier to use #1836 to use github submission on different versions of `autolab/keys`).

![Screenshot 2023-03-26 at 16 40 32](https://user-images.githubusercontent.com/9074856/227803391-6473da5e-57de-4fcb-8d0a-263e626364f2.png)
- Ensure that dropdown displays the versions containing the same file (excluding the current version)
- Ensure that dropdown updates when switching files (incl. cached files) via clicking
- Ensure that dropdown updates when switching files via clicking on annotations
- Ensure that clicking on a dropdown version brings you to the same file when the file exists, and the first file otherwise

Other things to test
- Previous / Next version buttons still work (bring you to the same file, in the previous / next version that contains the file)
- Download pdf / download annotated pdf still works

## Types of changes

## Checklist:
